### ISSUE TITLE
Optional Chaining is not checked properly?

### ISSUE THREAD

**Issue by Svish** [microsoft/TypeScript#36672]
**TypeScript Version:**  3.7.5

**Search Terms:** optional chaining return type 

**Code**

I have a formatting function, where the input should always be a string, but in certain cases, from the backend, it's `undefined` for some reason. So to make sure the front-end at least doesn't crash when that happens, I added optional chaining, which fixed the issue.

But the types are kind of off now, and I didn't realize until now because Typescript didn't complain about anything.

```ts
export const formatAccountNumber = (account: string): string =>
  account?.replace(/^(\d{4})(\d{2})(\d{5})/, '$1 $2 $3');
```

**Expected behavior:** Should get an error/warning about _something_ here. Preferably, I guess optional chaining should always expand the type to include `undefined` to whatever it is added to? Then, in this example, Typescript could say that the return type is `string`, but I'm trying to return `string | undefined`? Either that, or maybe it shouldn't be possible to use optional chaining on a value that does not include either the type `undefined` and/or `null`?

**Actual behavior:** No warnings or errors.

**Related Issues:** no

**Comment by nmain** [comment]
This is the first discussion point of https://github.com/microsoft/TypeScript/issues/36623.  Also see https://github.com/microsoft/TypeScript/issues/36163.

**Comment by awjreynolds** [comment]
Just to say this one caught us out in our codebase and took a while to figure out as it wasn't being picked up.
